docs/src/content/en/examples/memory/use-chat.mdx

```diff
@@ -72,31 +88,74 @@ export async function POST(request: Request) {
 
   // Handle cases where message might be null (e.g., initial load or error)
   if (!message || !message.content) {
-    // Return an appropriate response or error
     return new Response("Missing message content", { status: 400 });
   }
 
-  // Process with memory using the single message content
-  const stream = await agent.stream(message.content, {
+  // Get the agent from your Mastra Server
+  const agent = mastraClient.getAgent("ChatAgent"); // Use your agent's ID
+
+  // Stream the response with memory context
+  const response = await agent.stream({
+    messages: [{ role: message.role || "user", content: message.content }],
     threadId,
     resourceId,
-    // Pass other message properties if needed, e.g., role
-    // messageOptions: { role: message.role }
   });
 
-  // Return the streaming response
-  return stream.toDataStreamResponse();
+  // Return the streaming response to the frontend
+  return new Response(response.body);
```

Comment:
@greptileai The stream response handling is actually correct here. The MastraClient's `stream` method returns a Response object with a readable stream body, as documented in the MastraClient docs: https://github.com/mastra-ai/mastra/blob/main/docs/src/content/en/docs/server-db/mastra-client.mdx#L156-L158

This pattern (`new Response(response.body)`) is the recommended approach when using MastraClient on the server side to forward the stream to the client.